Generators/src/PrimaryGenerator.cxx

```diff
@@ -84,6 +85,14 @@ Bool_t PrimaryGenerator::GenerateEvent(FairGenericStack* pStack)
   mEmbedTree->GetEntry(mEmbedIndex);
   setInteractionVertex(mEmbedEvent);
 
+  /** notify event generators **/
+  auto genList = GetListOfGenerators();
+  for (int igen = 0; igen < genList->GetEntries(); ++igen) {
+    auto o2gen = dynamic_cast<Generator*>(genList->At(igen));
+    if (o2gen)
+      o2gen->notifyEmbedding(mEmbedEvent);
```

Comment:
shouldn't this be only done when we do embedding? Currently, it looks like it's always done.

Reply:
this is only done when we do embedding.
If no embedding the function returns before reaching this point